plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -771,32 +770,32 @@ private void updateView(ConnectorSession session, SchemaTableName viewName, Conn
     @Override
     public List<SchemaTableName> listMaterializedViews(ConnectorSession session, Optional<String> namespace)
     {
+        ImmutableList.Builder<SchemaTableName> materializedViews = ImmutableList.builder();
```

Comment:
> Use normal for loop in TrinoGlueCatalog.listMaterializedViews

seems like switching to a for loop isn't really necessary 

why not add additional catch condition to the `try catch` within `flatMap`?

Reply:
but there already was a catch. you need both in-loop and out-of-loop try-catches anyway, right?

Reply:
I thought you pointed another `flatMap(List::stream)`. The `AccessDeniedException` exception happened in the subsequent `collect()` method when I confirmed previously.

Reply:
code is worth thousand words... i pushed a commit showing what i have in mind.
please let me know wdyt

Reply:
> The AccessDeniedException exception happened in the subsequent collect() method when I confirmed previously.

This means the current your change didn't work. I will investigate once again.

Reply:
I reconfirmed the failed place was `computeNext()` method in `AwsSdkUtil#getPaginatedResults`. It causes the failure outside of the current catch block. 
```java
                        catch (EntityNotFoundException | AccessDeniedException e) {
                            // Namespace may have been deleted or permission denied
                            return Stream.empty();
                        }
                    })
                    .collect(toImmutableList()); // AccessDeniedException happened here
```
Sorry, I should have written more detailed information in the commit message.